eng/common/pipelines/templates/jobs/npm-publish.yml

```
@@ -1,18 +1,27 @@
 parameters:
-  Tag: 'latest'
+  Tag: 'auto'
   ArtifactName: 'packages'
+  ArtifactSubPath: ''
+  DeploymentName: 'PublishPackage'
   DependsOn: []
   Environment: 'package-publish'
   Registry: 'https://registry.npmjs.org/'
+  Pool: # hardcoding the pool and image name because deployment jobs do not support variable expansion in pool names
+    name: azsdk-pool
+    image: ubuntu-24.04
+    os: linux
+  CustomCondition: succeeded()
+  FailOnMissingPackages: true
 
 jobs:
-- deployment: PublishPackage_${{ parameters.ArtifactName }}
-  displayName: 'Publish ${{ parameters.ArtifactName }} to ${{ parameters.Registry }}'  
+- deployment: ${{ parameters.DeploymentName }}
+  displayName: 'Publish ${{ parameters.ArtifactName }} to ${{ parameters.Registry }}' 
+  condition: ${{ parameters.CustomCondition }}
```

```suggestion
  condition: and(not(canceled()), ${{ parameters.CustomCondition }})
```

just to cancel-proof custom conditions

---

For now, I'm going to let the consumer control the full condition. If we decide to go with this pattern we can make a larger pass.